### ISSUE TITLE
code block highlighting with lexer specified

### ISSUE THREAD

**Issue by madjxatw** [marshallward/vim-restructuredtext#45]
Code block with lexer specified explicitly, the asterisk is regarded as a RST markup, resulting in the subsequent text being rendered in italic.

![vim-rst-codeblock-lexer](https://user-images.githubusercontent.com/6035833/68859894-4587d100-0723-11ea-83e8-586c8471e563.png)

Wihtout lexer specified, it works as expected.

![vim-rst-codeblock](https://user-images.githubusercontent.com/6035833/68859889-40c31d00-0723-11ea-8069-b3d739672d45.png)

This might be related to #40 , does the fix for it introduce the new issue?

**Comment by marshallward** [comment]
Somehow I completely missed that this has been posted, very sorry about that.  I can reproduce this problem, and will see how it's related to #40

**Comment by madjxatw** [comment]
When first encountering this issue, I felt pretty confused about why the `code-block::` directive with an argument specified produced even worse result than that without argument. I managed to figure it out by reading the discussion in #40–something had been changed to particularly handle the no-argument case. My problem is not tied to #40, but the fix for #40 might increase more confusion indeed.

I think the reporter of bug #40 might confuse the docutils directives with Sphinx directives. Sphinx `code-block::` directive doesn't support no-argument form while the docutils `code::` directive does. Your fix for #40 had better target only at `code::` directive. For `code-block::` given without
argument, its body content should be kept in regular text rather than literal block, thus altering users to the missing argument. 

My problem is actually with the language fallback. The `console` is a valid lexer in pygments (the syntax highlighter used by Sphinx) but not in the `g:rst_syntax_code_list`, therefore nothing is to match against `.. sourcecode:: console`. As I suggested in #40, any language that is not listed in `g:rst_syntax_code_list` should fallback to literal block.
